INTMDB-370 -- PIT Restore Example (#870)
mongodb/terraform-provider-mongodbatlas: website/docs/r/cloud_backup_snapshot_restore_job.html.markdown

-> **API Key Access List**: This resource requires an Atlas API Access Key List to utilize this feature. This means to manage this resources you must have the IP address or CIDR block that the Terraform connection is coming from added to the Atlas API Key Access List of the Atlas API key you are using. See [Resources that require API Key List](https://www.mongodb.com/docs/atlas/configure-api-access/#use-api-resources-that-require-an-access-list) for details.
## Example Usage

> ### Examples are hosted alongside the provider source code:
> - A `v1.1.0` compatible [example](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/examples/test-upgrade/v110/cloud-backup-snapshot/v110) with automated backup restore on AWS
> - A `v1.4.6` compatible [example](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/examples/test-upgrade/v146) with point in time restore on AWS.

Reviewer: Great to have the examples and even better idea to link to them from the docs (not sure why we haven't thought of this before) but we still need a section here with something like "### Example of a point in time restore" and a block like the examples below (e.g. line 25/29) with content from examples/test-upgrade/v146/cloud-backup-pit-restore/main.tf.   Then for the links to the full examples we could turn this into an entry at the end.  something like this, with updated locations.  @Zuhairahmed we could potentially do this for other areas of the docs.  We need to ensure examples get constantly updated to do this but it could really help users find them.
```suggestion
### Available Complete Examples
- A `v1.1.0` compatible [example](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/examples/test-upgrade/v110/cloud-backup-snapshot/v110) with automated backup restore on AWS
- A `v1.4.6` compatible [example](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/examples/test-upgrade/v146) with point in time restore on AWS.
```

Reviewer 2: agree could be worthwhile, created internal ticket [INTMDB-445 ](https://jira.mongodb.org/browse/INTMDB-445) to explore further

  }
```

> ## Argument Reference

Reviewer: We need to also better explain below that if * `delivery_type_config.pointInTime` is set to true then one must also use one of the **pointInTime** optional values.  I think that's what tripped up the reporter in the ticket INTMDB-370.
